neuralop/models/mwt.py

```diff
@@ -0,0 +1,666 @@
+import torch
+import numpy as np
+import torch.nn as nn
+import torch.nn.functional as F
+from torch import Tensor
+from typing import List, Tuple, Union, Literal
+import math
+
+from functools import partial
+from scipy.special import eval_legendre
+from sympy import Poly, legendre, Symbol, chebyshevt
+
+class WaveletUtils:
+    """Unified wavelet utility class containing wavelet transform functions for all dimensions"""
```

Comment:
Some docs here would be helpful. Please refer to the contributors guide for guidance on how to format in-line docstrings